clang/include/clang/Parse/ParseHLSLRootSignature.h

```diff
@@ -80,6 +85,99 @@ class RootSignatureLexer {
   }
 };
 
+class RootSignatureParser {
+public:
+  RootSignatureParser(SmallVector<rs::RootElement> &Elements,
+                      const SmallVector<RootSignatureToken> &Tokens,
+                      DiagnosticsEngine &Diags);
+
+  // Iterates over the provided tokens and constructs the in-memory
+  // representations of the RootElements.
+  //
+  // The return value denotes if there was a failure and the method will
+  // return on the first encountered failure, or, return false if it
+  // can sucessfully reach the end of the tokens.
```

Comment:
If you use `///` instead of `//` for the doc comments on the methods of this class they'll show up in the [LLVM doxygen](https://llvm.org/doxygen/).